src/test/blockchain/blockchain_behavior_tests.cpp

```diff
@@ -39,4 +49,92 @@ BOOST_AUTO_TEST_CASE(creation_test) {
   }
 }
 
+BOOST_AUTO_TEST_CASE(GetTransactionWeight_test) {
+  mocks::ArgsManagerMock args{};
+  const auto b = blockchain::Behavior::New(&args);
+
+  {
+    // Check weight of empty transaction
+    CTransaction tx;
+
+    const std::size_t expected = GetTransactionWeight(tx);
+    const std::size_t weight = b->GetTransactionWeight(tx);
+
+    BOOST_CHECK_EQUAL(40, weight);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+
+  txtools::TxTool key_tool;
+  {
+    // Check weight of standard transaction
+    const CTransaction tx = key_tool.CreateTransaction();
+
+    const std::size_t expected = GetTransactionWeight(tx);
+    const std::size_t weight = b->GetTransactionWeight(tx);
+
+    BOOST_CHECK_EQUAL(437, weight);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(GetBlockWeight_test) {
+  mocks::ArgsManagerMock args{};
+  const auto b = blockchain::Behavior::New(&args);
+
+  {
+    // Check weight of empty block
+    CBlock block;
+
+    const std::size_t expected = GetBlockWeight(block);
+    const std::size_t weight = b->GetBlockWeight(block);
+
+    BOOST_CHECK_EQUAL(568, weight);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+
+  txtools::TxTool key_tool;
+  {
+    // Check weight of some block
+    CBlock block;
+    const CTransaction tx = key_tool.CreateTransaction();
+    block.vtx.emplace_back(MakeTransactionRef(tx));
+
+    const std::size_t expected = GetBlockWeight(block);
+    const std::size_t weight = b->GetBlockWeight(block);
+
+    BOOST_CHECK_GE(weight, 1004);
+    BOOST_CHECK_LE(weight, 1005);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(GetTransactionInputWeight_test) {
+  mocks::ArgsManagerMock args{};
+  const auto b = blockchain::Behavior::New(&args);
+
+  {
+    // Check weight of empty transaction input
+    CTxIn txin;
+
+    const std::size_t expected = GetTransactionInputWeight(txin);
+    const std::size_t weight = b->GetTransactionInputWeight(txin);
+
+    BOOST_CHECK_EQUAL(165, weight);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+
+  txtools::TxTool key_tool;
+  {
+    // Check weight of minimal transaction input
+    const CTransaction tx = key_tool.CreateTransaction();
+    const CTxIn txin = tx.vin[0];
+    const std::size_t expected = GetTransactionInputWeight(txin);
+    const std::size_t weight = b->GetTransactionInputWeight(txin);
+
+    BOOST_CHECK_GE(weight, 270);
```

Comment:
I have repeated this block 10000000 times and it failed a few times:
`error: in "blockchain_behavior_test/GetTransactionInputWeight_test": check weight >= 270 has failed [269 < 270]`
I guess all tests with signature size check are subject to this, because there is actually no strict lower limit on signature size.

However during this 10000000 repetitions I haven't seen values lower than 269, so it would be an acceptable compromise

Reply:
Changed in ad7a6ac49c223643ad3ce9d43f17d3ba49904e99